src/Test.UnitTests.Sarif.Multitool/ExtensionsTests.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool
+{
+    public class ExtensionsTests
+    {
+        [Theory]
```

Comment:
>Theory [](start = 9, length = 6)

We don't use `Theory` in this code base because we had some bad perf experience with it some years ago. It might be much better now. We use another technique for data-driven tests. I'll go find an example and add it to this comment. Stand by...

Reply:
For an example of our pattern, see src\Test.UnitTests.Sarif.WorkItems\SarifWorkItemExtensionsTests.cs, Lines 118-186.

---
In reply to: [455431411](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455431411) [](ancestors = 455431411)

Reply:
The set of test cases is fine. Please use the pattern in src\Test.UnitTests.Sarif.WorkItems\SarifWorkItemExtensionsTests.cs, Lines 118-186. This pattern accumulates failures across all test cases, and then displays them at the end. Also, use FluentAssertions.

---
In reply to: [455441780](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455441780) [](ancestors = 455441780,455431411)

Reply:
Ok, we are now defining the test data according to the pattern in src\Test.UnitTests.Sarif.WorkItems\SarifWorkItemExtensionsTests.cs, Lines 118-186. We still need to do the following:

- Accumulate the failing test cases. See how `SarifWorkItemExtensions_PhraseToolNames_ConstructPhraseCorrectly` accumulates a list of the failed test cases in a `StringBuilder`.
- Use FluentAssertions. See how that test method uses `sb.Length.Should().Be(0, "... message listing the failed test cases ...")`

---
In reply to: [455806704](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455806704) [](ancestors = 455806704,455441780,455431411)

Reply:
The reason for accumulating the test cases is that you can report all the failures at once, instead of stopping at the first failure.

---
In reply to: [455835254](https://github.com/microsoft/sarif-sdk/pull/1995#discussion_r455835254) [](ancestors = 455835254,455806704,455441780,455431411)